### PR TITLE
chore: add missing Authenticator text to default

### LINKED ISSUE #3510: Authenticator TextField placeholders are not translated
All Authenticator text placeholders are missing from translation files. Example: `Enter your Confirmation Code`  https://github.com/aws-amplify/amplify-ui/blob/270d7553652592d0d419ee708043b8292c63336c/packages/ui/src/helpers/authenticator/constants.ts#L18 https://github.com/aws-amplify/amplify-ui/blob/270d7553652592d0d419ee708043b8292c63336c/packages/ui/src/i18n/dictionaries/authenticator/en.ts#L3

Originally posted by @philipheinser in https://github.com/aws-amplify/amplify-ui/issues/3444#issuecomment-1457088341

### PR DESCRIPTION
**Pull request by zchenwei**
#### Description of changes

[Placeholders](https://github.com/aws-amplify/amplify-ui/blob/270d7553652592d0d419ee708043b8292c63336c/packages/ui/src/helpers/authenticator/constants.ts#L11) are missing from translation files. Adding them to the `DefaultText` so that they are discoverable.

#### Issue #, if available

fixes #3510

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by 0618 — DISMISSED** [review]

**Review by reesscot — DISMISSED** [review]
LGTM, thanks for fixing. Can you please also put "fixes" in front of the issue in your description so that it's automatically closed?

**Comment by zchenwei** [comment]
> LGTM, thanks for fixing. Can you please also put "fixes" in front of the issue in your description so that it's automatically closed?

Good call. I forgot this magic :(

**Review by wlee221 — COMMENTED** [review]
Can you add these to https://github.com/aws-amplify/amplify-ui/blob/9a257c7efccd7f3f0ed0c21beda4f7645a622de0/packages/ui/src/i18n/dictionaries/authenticator/en.ts too?

**Comment by reesscot** [comment]
> Can you add these to https://github.com/aws-amplify/amplify-ui/blob/9a257c7efccd7f3f0ed0c21beda4f7645a622de0/packages/ui/src/i18n/dictionaries/authenticator/en.ts too?

Oh right, I guess we are missing adding it to the english dictionary. Once it's there, we can ask for translations from contributors.

**Review by 0618 — APPROVED** [review]

**Review by wlee221 — APPROVED** [review]

**Review by zchenwei — COMMENTED** [review]
Updating types will break linting since we are missing other translations. Removed it

**Review by reesscot — DISMISSED** [review]

File: .changeset/new-meals-serve.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@aws-amplify/ui": patch
+---
+
+chore: add missing Authenticator text placeholders to deafult text
```
Comment (reesscot) [inline .changeset/new-meals-serve.md:5]:
```suggestion
chore: add missing Authenticator text placeholders to default authenticator text and `en` dictionary
```

**Review by ioanabrooks — APPROVED** [review]

File: packages/ui/src/i18n/dictionaries/authenticator/defaultTexts.ts [hunk 2]
```diff
@@ -21,7 +24,18 @@ export const defaultTexts = {
   CREATE_ACCOUNT: 'Create Account',
   CREATING_ACCOUNT: 'Creating Account',
   EMAIL_ADDRESS: 'Email',
+  ENTER_BIRTHDATE: 'Enter your Birthdate',
```
Comment (reesscot) [inline packages/ui/src/i18n/dictionaries/authenticator/defaultTexts.ts:27]: Feels weird that we duplicate this text, I wonder if we should just pull from the `en` dictionary?

**Review by reesscot — APPROVED** [review]

File: packages/ui/src/i18n/dictionaries/authenticator/defaultTexts.ts (hunk 2 above)
Comment (zchenwei) [reply packages/ui/src/i18n/dictionaries/authenticator/defaultTexts.ts:27]: I feel the same. Or maybe vice verse :)

**Pull request merged**